asn1/CommonLedgerProtocol.asn

```diff
@@ -0,0 +1,114 @@
+CommonLedgerProtocol
+DEFINITIONS
+AUTOMATIC TAGS ::=
+BEGIN
+
+IMPORTS
+    UInt8,
+    UInt32,
+    UInt64,
+    UInt128,
+    UInt256
+    FROM GenericTypes
+
+    Timestamp
+    FROM InterledgerTypes
+
+    InterledgerPacket
+    FROM InterledgerPacket
+;
+
+-- Zero or more protocolData entries, each with a unique
+-- protocolName, describe protocol-specific information about
+-- unpaid and conditionally paid requests and results
+-- sent over the CommonLedgerProtocol.
+-- The protocolName of each entry describes the protocol for
+-- interpreting the data. Peers need to establish beforehand
+-- which protocols each understand, and which names they
+-- use, which can be agreed ad-hoc, except that protocolName 'ilp'
+-- is reserved for the Interledger protocol.
+-- When multiple protocolData items are sent in a request,
+-- only the first one will be decisive in triggering the
+-- response, the rest of the protocolData items are considered
+-- side protocols that piggyback on the first one.
+-- The contentType byte can be used to allow debug tools to
+-- display the contents of the protocolData without understanding
+-- it.
+
+ContentType ::= INTEGER {
```

Comment:
Can this just be an `IA5String` so we don't have CLP re-defining Content-Types?

Reply:
> This field is intended to give applications that don't know/understand the custom protocol a way to at least partially decode the data.

Ok, perhaps I've been viewing this field a bit differently. I was thinking this content-type would be useful for applications that _do_ understand the side-protocol...i.e., it gives those applications a mechanism to decode the data. For example, if I have a side-protocol named `aml_protocol`, then perhaps this payload is encoded as JSON, perhaps it's binary. Are you suggesting that if `aml_protocol` wants different encodings, then one should create separate protocolNames, like `aml_protocol_json` and `aml_protocol_oer`?